### ISSUE TITLE
zookeper.restart can throw exceptions

### ISSUE THREAD

**Issue by ov7a** [Parsely/pykafka#94]
Code (Main Thread):

    while True:
        try:
            for message in consumer:
                #do smth
                consumer.commit_offsets()
        except (SessionExpiredError, ConnectionLoss):
            zookeeper.restart()

Some troubles with network -> consumer throw an Exception -> zookeeper restarting -> failed to restart -> another exception managed by no one -> everything fails. 

Exception:

    Traceback (most recent call last):
      File "worker.py", line 313, in <module>
        zookeeper.restart()		  
      File "/usr/local/lib/python2.7/site-packages/kazoo-1.1-py2.7.egg/kazoo/client.py", line 474, in restart
        self.stop()
      File "/usr/local/lib/python2.7/site-packages/kazoo-1.1-py2.7.egg/kazoo/client.py", line 470, in stop
        self._safe_close()
      File "/usr/local/lib/python2.7/site-packages/kazoo-1.1-py2.7.egg/kazoo/client.py", line 377, in _safe_close
        "and wouldn't close after %s seconds" % timeout)
    Exception: Writer still open from prior connection and wouldn't close after 10 seconds

So, basically, there is no way to keep connection persistent?
